crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -233,7 +233,7 @@ impl<T: Schedulable<Metadata = GraphInfo, GroupMetadata = Chain>> ScheduleConfig
     ///
     /// This is useful if you have a run condition whose concrete type is unknown.
     /// Prefer `run_if` for run conditions whose type is known at compile time.
-    pub fn run_if_dyn(&mut self, condition: BoxedCondition) {
+    pub fn run_if_dyn_inner(&mut self, condition: BoxedCondition) {
```

Comment:
I don't understand why this needs to exist. Can't you just discard the unused ScheduleConfigs?